NET_project/Benchmarks/PixarRaytracer.cs

```diff
@@ -17,150 +17,125 @@ public unsafe struct PixarRaytracerNET : IJob
+	private static float PixarRaytracer(uint width, uint height, uint samples)
 	{
 		marsagliaZ = 666;
 		marsagliaW = 999;
 
-		Vector3 position = new Vector3 { X = -22.0f, Y = 5.0f, Z = 25.0f };
-		Vector3 goal = new Vector3 { X = -3.0f, Y = 4.0f, Z = 0.0f };
+		Vector3 position = new Vector3(-22.0f, 5.0f, 25.0f);
+		Vector3 goal = new Vector3(-3.0f, 4.0f, 0.0f);
 
-		goal = Vector3.Add(Inverse(goal), Vector3.Multiply(position, -1.0f));
+		goal = Vector3.Normalize(goal) - position;
 
-		Vector3 left = new Vector3 { X = goal.Z, Y = 0, Z = goal.X };
+		Vector3 left = new Vector3(goal.Z, 0, goal.X);
 
-		left = Vector3.Multiply(Inverse(left), 1.0f / width);
+		left = Vector3.Normalize(left) / width;
 
 		Vector3 up = Vector3.Cross(goal, left);
-		Vector3 color = default(Vector3);
-		Vector3 adjust = default(Vector3);
+		Vector3 color = Vector3.Zero;
+		Vector3 adjust;
 
-		for (uint y = height; y > 0; y--)
-		{
-			for (uint x = width; x > 0; x--)
-			{
-				for (uint p = samples; p > 0; p--)
-				{
-					color = Vector3.Add(color, Trace(position, Vector3.Add(Inverse(Vector3.Multiply(Vector3.Add(goal, left), x - width / 2 + Random())), Vector3.Multiply(up, y - height / 2 + Random()))));
+		for (uint y = height; y > 0; y--) {
+			for (uint x = width; x > 0; x--) {
+				for (uint p = samples; p > 0; p--) {
+					color += Trace(position, Vector3.Normalize((goal + left) * (x - (width / 2) + Random())) + (up * (y - (height / 2) + Random())));
 				}
 
-				color = Vector3.Multiply(color, (1.0f / samples) + 14.0f / 241.0f);
-				adjust = AddFloat(color, 1.0f);
-				color = new Vector3 {
-					X = color.X / adjust.X,
-					Y = color.Y / adjust.Y,
-					Z = color.Z / adjust.Z
-				};
+				color *= (1.0f / samples) + (14.0f / 241.0f);
+				adjust = color + Vector3.One;
+				color /= adjust;
 
-				color = Vector3.Multiply(color, 255.0f);
+				color *= 255.0f;
 			}
 		}
 
 		return color.X + color.Y + color.Z;
 	}
 
 	[MethodImpl(MethodImplOptions.AggressiveInlining)]
-	private Vector3 Inverse(Vector3 vector) {
-		return Vector3.Multiply(vector, 1 / (float)Math.Sqrt(Vector3.Dot(vector, vector)));
-	}
-
-	[MethodImpl(MethodImplOptions.AggressiveInlining)]
-	private Vector3 AddFloat(Vector3 vector, float value) {
-		vector.X += value;
-		vector.Y += value;
-		vector.Z += value;
-
-		return vector;
-	}
-
-	[MethodImpl(MethodImplOptions.AggressiveInlining)]
-	private float Min(float left, float right) {
+	private static float Min(float left, float right) {
 		return left < right ? left : right;
 	}
 
 	[MethodImpl(MethodImplOptions.AggressiveInlining)]
-	private float BoxTest(Vector3 position, Vector3 lowerLeft, Vector3 upperRight) {
-		lowerLeft = Vector3.Multiply(Vector3.Add(position, lowerLeft), -1);
-		upperRight = Vector3.Multiply(Vector3.Add(upperRight, position), -1);
+	private static float BoxTest(Vector3 position, Vector3 lowerLeft, Vector3 upperRight) {
+		lowerLeft = -(position + lowerLeft);
```

Comment:
There's a "lot" of C code that is multiplying by `-1` as a means to negate.

It's something that C/C++ compilers may transform to a negation themselves, but which isn't actually "safe" for regular code to do per IEEE 754 rules as it can cause incorrect behavior for some types of `NaN`.

Simply doing negation/subtraction is more idiomatic anyways and none of these are `NaN`, so I manually applied that fix.